### PR TITLE
[CS-3887] Refactor SlackSheet to use /cardstack Sheet component

### PR DESCRIPTION
**Pull request by paulinhapenedo**
TL:DR;
This PR removes a lot of unused code and moves code that was already ported from JS to TS out of rainbow and into `/cardstack` folder.

**Longer version**
- `SlackSheet` component is gone and the components that used it were ported to TS and moved into `/cardstack`.
- Removed a bunch of unused files.
- Removed the `EXPANDED_ASSET_SHEET` and `EXPANDED_ASSET_SHEET_DRILL` routes in favor of using the direct routes to the components.
- Moved all chart related components into the Charts folder (`/value-charts`). They were in the `/sheets` folder and it didn't make sense since there was no chart component inside the `/sheets` folder anymore.

- [x] Completes #CS-3887

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

The only visual change is the "Buy Prepaid Card" screen that's now following the design specs.
<image width="350" src="https://user-images.githubusercontent.com/690904/170589314-5372b7ef-191b-478f-bf19-4bc23b81fdbf.png" />

### CODE REVIEW

File: cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx [hunk 1]
```diff
@@ -64,95 +58,81 @@ const BuyPrepaidCard = () => {
...
-                justifyContent="center"
-              >
-                <Text color="white" marginRight={1}>
-                  Works with most debit cards
-                </Text>
-                <Icon name="info" size={15} />
-              </Container>
-            </Touchable>
+            <Text color="white" marginRight={1}>
+              Works with most debit cards
```
Comment (danibonilha) [inline cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx]: nit: it would be nice to extract the strings to a strings files

File: cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx [hunk 2]
```diff
@@ -64,95 +58,81 @@ const BuyPrepaidCard = () => {
...
+      cardBackgroundColor={colors.backgroundBlue}
+    >
+      <Container backgroundColor="backgroundBlue" paddingHorizontal={5}>
+        <TopContent />
+        <Subtitle text="CHOOSE AMOUNT" />
+        <FlatList
+          data={inventoryData}
+          renderItem={renderItem}
+          numColumns={2}
+          keyExtractor={(item, index) => index.toString()}
```
Comment (danibonilha) [inline cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx]: nit: extract to it's own callback function and add a comment that using the index is necessary to avoid duplicated loading skeletons

File: cardstack/src/screens/sheets/SupportAndFeedsSheet/SupportAndFeedsSheet.tsx [hunk 3]
```diff
@@ -1,32 +1,16 @@
...
-        info: '$1,000 per week  max equivalent, $7,500 yearly max equivalent',
-      },
-    },
-    footerInfo: `Buy Prepaid Card is not available in NY & TX\nWyre currently only supports Visa or Mastercard\n\nApple Cash Cards are not supported at this time`,
-  },
-  supportedCountriesTitle: 'Supported Countries',
-};
-
-const SupportAndFeedsState = () => {
+const SupportAndFeedsSheet = () => {
```
Comment (danibonilha) [inline cardstack/src/screens/sheets/SupportAndFeedsSheet/SupportAndFeedsSheet.tsx]: oops, it looks like we had a typo all this time 😅  `SupportAndFees`

File: cardstack/src/screens/sheets/TokenWithChartSheet/TokenWithChartSheet.tsx [hunk 4]
```diff
@@ -1,32 +1,30 @@
...
 
-  const asset = props.asset?.token
-    ? {
-        ...props.asset,
-        ...props.asset.token,
-      }
-    : currentAsset;
+export default function TokenWithChartSheet() {
+  const { params } = useRoute<
+    RouteType<{ asset: AssetWithNativeType; safeAddress?: string }>
```
Comment (danibonilha) [inline cardstack/src/screens/sheets/TokenWithChartSheet/TokenWithChartSheet.tsx]: superNit: We could extract the interface to it's own

File: cardstack/src/screens/sheets/TokenWithChartSheet/components/AmountWithCoin.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { Container, CoinIcon, Text } from '@cardstack/components';
+
+import { AmountRow } from '../types';
+
+export default function AmountWithCoin({ title, asset }: AmountRow) {
+  return (
```
Comment (danibonilha) [inline cardstack/src/screens/sheets/TokenWithChartSheet/components/AmountWithCoin.tsx:8]: nit: we could convert it to arrow function and export default wrapped with `memo`

Comment (danibonilha) [inline cardstack/src/screens/sheets/TokenWithChartSheet/components/AmountWithCoin.tsx:8]: nit: we could convert it to arrow function and export default wrapped with `memo`

File: cardstack/src/screens/sheets/TokenWithChartSheet/components/NativeAmount.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    <Container>
+      <Text color="grayText" fontSize={13} marginBottom={1} textAlign="right">
+        {title}
+      </Text>
+      <Text fontSize={20} weight="extraBold">
+        {asset?.native?.balance.display}
+      </Text>
+    </Container>
+  );
+}
```
Comment (danibonilha) [inline cardstack/src/screens/sheets/TokenWithChartSheet/components/NativeAmount.tsx]: ditto

File: cardstack/src/screens/sheets/TokenWithChartSheet/types.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { AssetWithNativeType } from '@cardstack/types';
+
+export interface AmountRow {
```
Comment (danibonilha) [inline cardstack/src/screens/sheets/TokenWithChartSheet/types.ts]: nit, maybe `AmountRowProps`

File: src/components/value-chart/chart-data-labels/ChartChangeDirectionArrow.js [hunk 8]
```diff
@@ -5,8 +5,9 @@ import Animated, {
   useDerivedValue,
 } from 'react-native-reanimated';
 import styled from 'styled-components';
-import { Icon } from '../../../icons';
+
 import { useRatio } from './useRatio';
+import { Icon } from '@rainbow-me/components/icons';
 
 const AnimatedMaskedView = Animated.createAnimatedComponent(MaskedView);
```
Comment (danibonilha) [inline src/components/value-chart/chart-data-labels/ChartChangeDirectionArrow.js:13]: nit: we might not even need this component, as it's a conditional to android only, but the PercentLabel is enough

**Review by danibonilha — APPROVED** [review]
This makes me so happy! Nice job, just some small nits

File: cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx [hunk 9]
```diff
@@ -64,95 +58,81 @@ const BuyPrepaidCard = () => {
...
-            <TopContent />
-          </Container>
+  const renderFooter = useMemo(
+    () => (
+      <Container paddingHorizontal={2} justifyContent="center">
+        {isDisabled ? null : (
+          <ApplePayButton
+            disabled={isPurchaseInProgress}
+            onSubmit={handlePurchase}
+            onDisabledPress={() => console.log('onDisablePress')}
```
Comment (oidoug) [inline cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx]: nit: This log could be removed, I think it's only there bc ApplePayButton doesn't check if onDisabledPress is defined before calling it. Adding `onDisabledPress?.()` on ApplePayButton would be enough.

File: src/components/value-chart/chart-data-labels/ChartChangeDirectionArrow.js (hunk 8 above)
Comment (paulinhapenedo) [reply src/components/value-chart/chart-data-labels/ChartChangeDirectionArrow.js:13]: Should I remove it from here too?

```
<AnimatedMaskedView
        maskElement={<ArrowIcon />}
        style={{ height: 18, width: 15 }}
      >
```

File: cardstack/src/screens/sheets/TokenWithChartSheet/components/AmountWithCoin.tsx (hunk 5 above)
Comment (oidoug) [reply cardstack/src/screens/sheets/TokenWithChartSheet/components/AmountWithCoin.tsx:8]: +1 (on updating it to an arrow fn and memorizing).

**Review by oidoug — APPROVED** [review]
LGTM. Good job Paula!

File: cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx (hunk 9 above)
Comment (paulinhapenedo) [reply cardstack/src/screens/BuyPrepaidCard/BuyPrepaidCard.tsx]: Done! Thanks for the heads-up!
